src/context.rs

```diff
@@ -933,57 +986,24 @@ impl SeafowlContext for DefaultSeafowlContext {
-                        // thus polluting e.g. the information_schema output), or even worse reloading
-                        // the map and having the versioned query fail during execution.
-                        let session_ctx = SessionContext::with_state(self.inner.state());
-
-                        for (table, version) in &version_processor.table_versions {
-                            let name_with_version =
-                                TableVersionProcessor::table_with_version(table, version);
 
-                            let full_table_name = table.to_string();
-                            let mut resolved_ref = TableReference::from(full_table_name.as_str()).resolve(&self.database, DEFAULT_SCHEMA);
-
-                            // We only support datetime DeltaTable version specification for start
-                            let table_uuid = self.get_table_uuid(resolved_ref.clone()).await?;
-                            let table_object_store =
-                                self.internal_object_store.for_delta_table(table_uuid);
-                            let datetime = TableVersionProcessor::version_to_datetime(version)?;
-
-                            let mut delta_table = DeltaTable::new(table_object_store, Default::default());
-                            delta_table.load_with_datetime(datetime).await?;
-                            let table_provider_for_version = Arc::from(delta_table);
-
-                            resolved_ref.table = Cow::Borrowed(name_with_version.as_str());
-
-                            if !session_ctx.table_exist(resolved_ref.clone())? {
-                                session_ctx.register_table(resolved_ref, table_provider_for_version)?;
-                            }
-                        }
-
-                        let state = session_ctx.state();
-                        return state.statement_to_plan(DFStatement::Statement(Box::from(Statement::Query(q)))).await;
-                    }
-
-                    state.statement_to_plan(DFStatement::Statement(Box::from(Statement::Query(q)))).await
-                },
+        // Create a mutable clone of the statement so that we can rewrite table names if we encounter
+        // time travel syntax.
+        // Alternatively, this could be done without the `mut`, except then we'd need to construct
+        // and return a new `DFStatement` after the rewrite. In case of `Statement::Query` this is
+        // straight-forward as that enum variant contains a struct, however `Statement::Insert` and
+        // `Statement::CreateTable` wrap dozens of fields in curly braces, and we'd need to capture
+        // and re-populate each of those, so it would be very verbose.
+        // TODO: If `sqlparser-rs` encapsulates these fields inside a struct at some point remove
+        // the mut and go with @ capture and de-structuring to pass along other fields unchanged.
+        let mut stmt = statement.clone();
```

Comment:
We already `clone`d the statement so that wont incur a time penalty. In theory the `mut` might, but I think it's negligible if any, given that it's local to a single function.

The alternative I mention is to have things as before and then capture and re-create a enum value with all the fields explicitly (since @ capture doesn't work in this case), i.e. https://github.com/sqlparser-rs/sqlparser-rs/blob/main/src/ast/mod.rs#L1178-L1198 for insert (even more verbose for create table).